core/src/Network/AWS/Data/Query.hs

```diff
@@ -22,6 +23,7 @@ module Network.AWS.Data.Query where
 import           Data.ByteString.Builder     (Builder)
 import qualified Data.ByteString.Builder     as Build
 import qualified Data.ByteString.Char8       as BS8
+import           Data.ByteString.Internal    (ByteString (..))
```

Comment:
This will need CPP around it as well

Reply:
For simplicity, decided to just drop the use if the `Internal` module and use `BS8.length` instead.